beamer/tests/agent/test_challenge.py

```diff
@@ -408,3 +482,120 @@ def test_challenge_7(request_manager, fill_manager, token, config):
 
     assert agent_earnings() == stake
     assert dave_earnings() == stake + 1
+
+
+# Scenario:
+#
+# Bob          Charlie
+# --------------------------
+#                  dishonest claim
+# honest claim
+# challenge
+#                  challenge
+# Winner: Charlie
+def test_challenge_l1_resolved_as_challenger(
+    request_manager, token, config, direction, finality_period
+):
+    requester, target = alloc_accounts(2)
+    (charlie,) = alloc_whitelisted_accounts(1, [request_manager])
+    agent = beamer.agent.agent.Agent(config)
+
+    ape.accounts.test_accounts[0].transfer(charlie, ape.convert("5 ether", int))
+
+    request_id = make_request(request_manager, token, requester, target, 1)
+
+    stake = request_manager.claimStake()
+    request_manager.claimRequest(request_id, FILL_ID, sender=charlie, value=stake)
+
+    collector = EventCollector(request_manager, "ClaimMade")
+
+    # charlie claim
+    claim_event = collector.next_event()
+    assert claim_event is not None
+
+    agent.start()
+
+    with Sleeper(10) as sleeper:
+        while (request := agent.get_context(direction).requests.get(request_id)) is None:
+            sleeper.sleep(0.1)
+
+    with Sleeper(10) as sleeper:
+        while (claim := agent.get_context(direction).claims.get(claim_event.claimId)) is None:
+            sleeper.sleep(0.1)
+
+    # honest fill and claim for the request by agent
+    claim_event = collector.next_event()
+    assert claim_event is not None
+
+    assert claim_event.claimId != claim.id
+
+    with freeze_time(
+        datetime.fromtimestamp(claim.invalidation_timestamp + finality_period + 1), tick=True
+    ):
+        # counter challenge with enough funds for l1 resolution for dishonest claim
+        claim_event = collector.next_event()
+        assert claim_event is not None
+
+        request_manager.challengeClaim(
+            claim_event.claimId, sender=charlie, value=ape.convert("3 ether", int)
+        )
+
+        ape.chain.mine(timestamp=claim.termination)
+
+        with Sleeper(20) as sleeper:
+            while not request.l1_resolved.is_active:
+                sleeper.sleep(0.1)
+
+    agent.stop()
+    agent.wait()
+
+
+# Scenario:
+#
+# Bob          Charlie
+# --------------------------
+# claim
+#                  challenge
+#
+# Winner: Charlie
```

Comment:
also here I assume Bob should be the winner

Reply:
as other pr had tests for resolution and invalidation i skipped these 2 and created another test to make sure we dont drain agent.